ansible_ai_connect/ai/api/model_pipelines/wca/tests/test_factory_onprem.py

```diff
@@ -49,6 +51,11 @@ def test_playbook_generation_pipeline(self):
             ModelPipelinePlaybookGeneration, WCAOnPremPlaybookGenerationPipeline
         )
 
+    def test_role_generation_pipeline(self):
+        self.assert_concrete_implementation(
+            ModelPipelineRoleGeneration, WCABaseRoleGenerationPipeline
```

Comment:
If you remove the class (as I propose above) this would need to check for the default NOP implementation.

Reply:
See my response above.